rcl_lifecycle/src/rcl_lifecycle.c

```diff
@@ -234,11 +257,12 @@ rcl_lifecycle_state_machine_init(
 rcl_ret_t
 rcl_lifecycle_state_machine_fini(
   rcl_lifecycle_state_machine_t * state_machine,
-  rcl_node_t * node_handle,
-  const rcl_allocator_t * allocator)
+  rcl_node_t * node_handle)
 {
   RCL_CHECK_ALLOCATOR_WITH_MSG(
-    allocator, "can't free state machine, no allocator given\n", return RCL_RET_INVALID_ARGUMENT);
+    &state_machine->options.allocator,
+    "Can't cleanup state machine. No allocator associated with the state machine.\n",
+    return RCL_RET_ERROR);
```

Comment:
checking the allocator doesn't seem necessary here, as the `init()` function already did that